DSpace/dspace-angular
# Fix Proxy Timeouts on browse by pages

## References
* Fixes #2282

## Description
This small PR fixes Proxy Timeouts on browse by pages when a non-existent browse index is given.

## Instructions for Reviewers
The `BrowseByGuard` only handled successful responses, now it will redirect you to the not found page instead

**Guidance for how to test or review this PR:**
- Go to `/browse/invalid` and check that you are redirected to the page not found page
- Check that the existing browse indexes like `/browse/author` still work

## Checklist
